preprocessing/test-data/regex_validator.py

```
@@ -0,0 +1,31 @@
+# - Risky regex patterns with catastrophic backtracking potential.
+# - Complex regex validation logic increasing cyclomatic complexity.
+# - Security risks due to unbounded input handling.
+
+import re
+
+def validate_email(email):
+    # Risk: Regex is vulnerable to catastrophic backtracking for malformed input
+    pattern = r'^([a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$'
+    if re.match(pattern, email):
+        return True
+    return False
```

Ruff (0.8.2)

Return the condition `bool(re.match(pattern, email))` directly

Replace with `return bool(re.match(pattern, email))`

(SIM103)